sdk/core/azure-core/azure/core/pipeline/policies/_retry.py

```diff
@@ -491,6 +497,8 @@ def send(self, request):
                 end_time = time.time()
                 if absolute_timeout:
                     absolute_timeout -= end_time - start_time
+        if not response:
```

Comment:
In what circumstances this would have happened?

Reading the code, this would replace an `AttributeError` on the line below with an `AzureError`, so I think this is great, but I'm curious to understand if we believe we discovered a bug, or if we think mypy is picky.

Reply:
If `response = self.next.send(request)` keeps raising error until maximum retries exceed, response will be None which is set in #L462.

I think this is a good catch by mypy.